core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala

```diff
@@ -252,8 +252,17 @@ class UncleanLeaderElectionTest extends ZooKeeperTestHarness {
 
     produceMessage(servers, topic, "third")
     waitUntilMetadataIsPropagated(servers, topic, partitionId)
-    servers.filter(server => server.config.brokerId == leaderId).map(server => shutdownServer(server))
+    //make sure follower server joins the ISR
+    TestUtils.waitUntilTrue(() => {
+      val partitionInfoOpt = followerServer.metadataCache.getPartitionInfo(topic, partitionId)
+      if (partitionInfoOpt.isDefined) {
```

Comment:
Could we just do
`partitionInfoOpt.isDefined && partitionInfoOpt.get.basePartitionState.isr.size == 2` for the boolean condition here?

Reply:
@abbccdda Thanks for the review. Updated the PR.